packages/viz/src/lib/sources/CARTOSource.ts

```diff
@@ -95,10 +96,108 @@ export class CARTOSource extends Source {
     return this._credentials;
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
-  // eslint-disable-next-line
-  public async getMetadataForField(field: string): Promise<Metadata> {
-    throw new Error('Method not implemented.');
+  public getFieldStats(): NumericFieldStats | CategoryFieldStats {
+    // initialize the stats to 0
+
+    if (!this.isInitialize)
+      throw new Error('To run this method you need to first call to init()');
+
+    if (this._fieldStats === undefined) {
+      throw new Error('Stats are not set after map instantiation');
+    }
+
+    return this._fieldStats;
+  }
+
+  private _initConfigForStats(field: string) {
+    if (this._mapConfig.metadata === undefined) {
+      throw new Error('Map Config has not metadata fiedl');
+    }
+
+    // Modify mapConfig to add the field stats
+    this._mapConfig.metadata.columnStats = {
+      topCategories: 32768,
+      includeNulls: true
+    };
+    this._mapConfig.metadata.dimensions = true;
+
+    /* eslint-disable @typescript-eslint/camelcase */
+    this._mapConfig.metadata.sample = {
+      num_rows: 1000,
+      include_columns: [field]
+    };
+    /* eslint-enable @typescript-eslint/camelcase */
+
+    this._mapConfig.aggregation = {
+      columns: {},
+      dimensions: {
+        [field]: { column: field }
+      },
+      placement: 'centroid',
+      resolution: 1,
+      threshold: 1
+    };
+  }
+
+  public async init(field?: string): Promise<boolean> {
+    if (this.isInitialize) {
+      // Maybe this is too hard, but I'd like to keep to check it's not a performance issue. We could move it to just a warning
+      throw new Error('Try to reinstantiate map multiple times');
+    }
+
+    const mapsClient = new Maps(this._credentials);
+
+    if (field) {
+      this._initConfigForStats(field);
+    }
+
+    const mapInstance: MapInstance = await mapsClient.instantiateMapFrom(
+      this._mapConfig
+    );
+
+    const urlData = mapInstance.metadata.url.vector;
+    const urlTemplate = urlData.subdomains.map((subdomain: string) =>
+      urlData.urlTemplate.replace('{s}', subdomain)
+    );
+
+    const { stats } = mapInstance.metadata.layers[0].meta;
+    this._geometryType = parseGeometryType(stats.geometryType);
+
+    this._layerProps = { type: 'TileLayer', data: urlTemplate };
+
+    // if (this._fieldStats !== undefined) {
+
+    if (field !== undefined) {
+      const columnStats = stats.columns[field];
+
+      switch (columnStats.type) {
+        case 'string':
+          this._fieldStats = {
+            name: field,
+            categories: columnStats.categories
```

Comment:
Are we getting here in categories both 'names' and 'number of elements' per category?

Reply:
Yes, both